segmentation/utils/dimensionality_reduction.py

```diff
@@ -0,0 +1,74 @@
+import pandas as pd
+import numpy as np
+from sklearn.preprocessing import StandardScaler
+from sklearn.decomposition import PCA
+from sklearn.manifold import TSNE
+import matplotlib.pyplot as plt
+import seaborn as sns
+import umap
+
+
+def visualize_dimensionality_reduction(cell_data, columns, category, color_map="Spectral", algorithm="UMAP",
+                                       save_dir=None):
+    """Plots the dimensionality reduction of specified population columns
+                Args:
+                    cell_data (pd df): Dataframe containing columns for dimensionality reduction and category
+                    columns (list): List of column names that are included for dimensionality reduction
+                    category (str): Name of column in dataframe containing categorical Population or Patient data
+                    color_map (str): Name of MatPlotLib ColorMap used, default is Spectral
+                    algorithm (str): Name of dimensionality reduction algorithm, default is UMAP
+                    save_dir (str): Directory to save plots, default is None"""
+    cell_data = cell_data.dropna()
+
+    assert(algorithm == "UMAP" or algorithm == "PCA" or algorithm == "tSNE")
```

Comment:
In this case, I'd raise a ValueError instead. Leave the assertions for the tests.

`if algorithm not in ["UMAP", "PCA", "tSNE"]: raise ValueError("The algorithm specified is not a valid dimensionality reduction technique")`